lib/web/brotlihandler.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package web
+
+import (
+	"io"
+	"mime"
+	"net/http"
+	"path"
+	"slices"
+	"strings"
+)
+
+var compressedFileExtensions = []string{
+	".js",
+	".svg",
+	".wasm",
+}
+
+// makeBrotliHandler serves pre-compressed .br files for supported file types.
+func makeBrotliHandler(handler http.Handler, fs http.FileSystem) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ext := path.Ext(r.URL.Path)
+		if !slices.Contains(compressedFileExtensions, ext) ||
+			!strings.Contains(r.Header.Get("Accept-Encoding"), "br") {
+			handler.ServeHTTP(w, r)
+			return
+		}
```

Comment:
I'd consider something like this for better readability:

```suggestion
		isRequestForCompressedFile := slices.Contains(compressedFileExtensions, ext)
		clientAcceptsBrotli := strings.Contains(r.Header.Get("Accept-Encoding"), "br")
		if !isRequestForCompressedFile || !clientAcceptsBrotli {
			handler.ServeHTTP(w, r)
			return
		}
```